### ISSUE TITLE
Allow appending multiple Set-Cookie Headers

### ISSUE THREAD

**Issue by nnelgxorz** [BuilderIO/qwik#1851]
### Is your feature request related to a problem?

The current headers implementation uses an append function that concatenates header values together with a ", ". This works for all headers except Set-Cookie.

### Describe the solution you'd like

There are a few options I can think of:

 - create special functions for each headers method specifically for cookies: `getCookie`, `setCookie`, `appendCookie`.
 - keep the existing headers methods, but check for the header value internally and store cookies separately. (if (name === 'Set-Cookie) {...}`
 - keep everything the same internally, but special case Set-Cookie to values on the ", " before creating the response and generating valid Set-Cookie headers for each value.
 - same as above, but use node-fetch's .raw() to get the Set-Cookie headers as an array.

### Describe alternatives you've considered

![cookie](https://user-images.githubusercontent.com/9634080/197366397-51cb8b2f-8300-4980-9ded-b119cf9c797a.jpg)

### Additional context

https://github.com/whatwg/fetch/issues/973 - Discussion around formalizing a solution for Set-Headers
https://github.com/withastro/astro/pull/3092/files - Astro PR using the adding multiple Set-Cookies for Netlify
https://github.com/sveltejs/kit/pull/3502/files - Svelte PR adding multiple Set-Cookies

**Comment by manucorporat** [comment]
I think good cookie support is super important, wanna take into preparing a PR most likely inspired by sveltekit and astro?

This would be such a massive improvement!!

**Comment by nnelgxorz** [comment]
Definitely! I was going to take it on yesterday, but it looks like Qwik City e2e tests are broken. I'm guessing because of the refactor Adam did on the middlewares/servers.

**Comment by nnelgxorz** [comment]
PR is ready for review!
